Package.swift

```diff
@@ -30,7 +30,7 @@ let package = Package(
             dependencies: [
                 .product(name: "NIO", package: "swift-nio"),
                 .product(name: "NIOCore", package: "swift-nio"),
-                .product(name: "NIOFoundationCompat", package: "swift-nio"),
+                .product(name: "NIOFoundationCompat", package: "swift-nio", condition: .when(platforms: [.iOS, .macOS, .tvOS, .watchOS, .macCatalyst, .driverKit])),
```

Comment:
We need `visionOS` here as well. That is introduced in a later tools version so we might need multiple `Package.swift` files here.

Reply:
Adding `#if compiler` check seems to work based on running `docker run --rm --volume .:/workspace swift:5.9.0 swift package resolve --package-path /workspace` and same with `swift:5.8`